### PR TITLE
Precompiled the regex of span/trace IDs

### PR DESCRIPTION
**Pull request by dianashevchenko**
### What does this PR do?
Replacing regex pattern strings from the parseTraceparent function with the precompiled patterns to avoid redundant allocations. 
The regex is checking whether the string is a valid hex.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Changed code has unit tests for its functionality.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

File: ddtrace/tracer/textmap.go [hunk 1]
```diff
@@ -686,6 +686,10 @@ var (
 	// Equals character must be encoded with a tilde.
 	// Other disallowed characters must be replaced with the underscore.
 	originRgx = regexp.MustCompile(",|~|;|[^\\x21-\\x7E]+")
+
+	// validIDRegex is used to verify that the input is a valid hex string.
+	// The input must match the pattern from start to end.
```
Comment (ahmed-mez) [inline ddtrace/tracer/textmap.go:691]: Let's be more specific about the input, i.e span ID? trace ID? both?

File: ddtrace/tracer/textmap.go [hunk 2]
```diff
@@ -847,7 +851,7 @@ func parseTraceparent(ctx *spanContext, header string) error {
 	if len(spanID) != 16 {
 		return ErrSpanContextCorrupted
 	}
-	if ok, err := regexp.MatchString("[a-f0-9]+", spanID); !ok || err != nil {
+	if ok := validIDRegex.MatchString(spanID); !ok {
```
[the remark this answers is not in the document]
Comment (dianashevchenko) [reply ddtrace/tracer/textmap.go:854]: Yes, and it is actually supposed to be from start to finish

File: ddtrace/tracer/textmap.go (hunk 1 above)
Comment (dianashevchenko) [reply ddtrace/tracer/textmap.go:691]: Both, will add `validIDRegex is applicable for both trace and span ID.`

File: ddtrace/tracer/textmap.go [hunk 3]
```diff
@@ -686,6 +686,11 @@ var (
 	// Equals character must be encoded with a tilde.
 	// Other disallowed characters must be replaced with the underscore.
 	originRgx = regexp.MustCompile(",|~|;|[^\\x21-\\x7E]+")
+
+	// validIDRegex is used to verify that the input is a valid hex string.
+	// The input must match the pattern from start to end.
+	// validIDRegex is applicable for both trace and span IDs.
+	validIDRegex = regexp.MustCompile("^[a-f0-9]+$")
```
Comment (ahmed-mez) [inline ddtrace/tracer/textmap.go:693]: Nit for naming consistency 😅 (all regex vars are named `<something>Rgx` in this file)
```suggestion
	validIDRgx = regexp.MustCompile("^[a-f0-9]+$")
```

Comment (dianashevchenko) [reply ddtrace/tracer/textmap.go]: Done

**Pull request merged**